src/main/kotlin/App.kt

```diff
@@ -1,32 +1,97 @@
+import com.amazonaws.services.lambda.runtime.Context
+import com.amazonaws.services.lambda.runtime.RequestHandler
 import com.google.api.client.util.DateTime
 import mu.KotlinLogging
 import java.io.File
+import java.text.SimpleDateFormat
 import java.util.*
 
 private val LOG = KotlinLogging.logger {}
 
+class Data {
+    var queryStringParameters: QueryStringParams? = QueryStringParams()
+
+    class QueryStringParams {
+        var computeRooms: Boolean = false
+    }
+}
+
+class AppHandler : RequestHandler<Data, String> {
+    override fun handleRequest(input: Data, context: Context?): String {
+        if (input.queryStringParameters != null) {
```

Comment:
instead of `!= null` u can use `input.queryStringParameters?.run { ... }` to insure `queryStringParameters` is not `null`